### PR TITLE
roachtest: remove mention of 'roachstress'

### PR DESCRIPTION
**Pull request by knz**
The proposed 'roachstress' script linked from the repro steps at
https://gist.github.com/tbg/fc4409e06b4ddc9abc58838cf5c0fe8b is
utterly broken: it confuses local with remote execution.

The correct steps must clarify that `roachprod` and `roachtest`
binaries must be built for the local platform, and the `cockroach` and
`workload` binaries must be built for the remote platform. Someone
trying to run this on macos, or with a linux without the old glibc,
gets inscrutable error messages with the current script.

Additionally, the proposed roachstress script uses a `--cpu-quota`
parameter that's incoherent with `roachtest`'s own default.

Finally, I (@knz) directly witnessed a new hire who was utterly
confused by the amount of automation and assumptions made in the
roachtest script. (Should they install caffeinate? Why? How to
customize the roachtest flags? How to change the number of runs? Can
the test run locally  on my computer?)

In summary, the `roachstress` script is just too magical and actively
hostile to newcomers.

This patch removes it and replaces it by clearer instructiosn.

Release note: None

### CODE REVIEW

**Comment by tbg** [comment]
Thanks Raphael.

The help text gets put into every issue and in particular into every email (where the details/summary tag isn't available), so the instructions should remain brief (i.e. a reference elsewhere)

I think at this point it makes sense to refer to the roachtest README, and to put instructions in there. Since I have a PR open against that anyway (https://github.com/cockroachdb/cockroach/pull/67656/files#diff-7858fe8937ab087da5c6be98767fa450680592d71173563c3caaee7f7d992c74R170) let me take that over.

**Comment by knz** [comment]
The `roachstress` is still incorrect.

If you want to instruct me how to make the explanation more brief, please do that. But we need to get rid of `roachstress`.

**Comment by tbg** [comment]
I was offering to replace the roachstress reference with a link into the README, but that can only be done when the README has been updated.

**Comment by knz** [comment]
I'll update the script with a link to the README then and then you can take it from there.

**Comment by knz** [comment]
I have added a simple linux-only command and a reference to the README.

**Review by tbg — APPROVED** [review]

**Comment by knz** [comment]
TFYR

bors r=tbg

**Comment by knz** [comment]
lovely

bors r=tbg
